AfrikaBurn/MapViewController.swift

```diff
@@ -0,0 +1,194 @@
+        mapView.showsUserLocation = true
+        mapView.delegate = self
+    
+        loadElements()
+        
+        self.navigationItem.title = "Map"
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        loadElements()
+    }
+    
+    func loadElements(){
+        
+        for element in allElements {
+            guard let location = element.location else {
+                continue
+            }
+            let anno = BurnAnnotation(coordinate: location);
+            anno.element = element
+            anno.title = element.name
+            anno.image = UIImage(named: "map-pin.png")
+            
+            mapView.addAnnotation(anno);
+        }
+    }
+    
+}
+
+class BurnMapView: MKMapView, MKMapViewDelegate {
+    
+    let overlay = BurnMapOverlay()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    func commonInit() {
+        showsScale = true
+        showsCompass = true
+        delegate = self
+        userTrackingMode = .followWithHeading
+        add(overlay, level: .aboveRoads)
+        setVisibleMapRect(BurnMap.boundingMapRect, animated: false)
+    }
+    func mapView(_ mapView: MKMapView, rendererFor overlay: MKOverlay) -> MKOverlayRenderer {
+        switch overlay {
+        case is BurnMapOverlay: return BurnMapOverlayView(overlay: overlay, overlayImage: Resources.mapOverlayImage)
+        default: return MKOverlayRenderer(overlay: overlay)
+        }
+    }
+}
+
+class BurnMapOverlay: NSObject, MKOverlay {
+    let boundingMapRect: MKMapRect = BurnMap.boundingMapRect
+    var coordinate: CLLocationCoordinate2D = CLLocationCoordinate2DMake(19.751633252193361, -32.326217923051772)
+    
+    func canReplaceMapContent() -> Bool {
+        return true
+    }
+}
+
+class BurnMapOverlayView: MKOverlayRenderer {
+    let overlayImage: UIImage
+    
+    init(overlay: MKOverlay, overlayImage: UIImage) {
+        self.overlayImage = overlayImage
+        super.init(overlay: overlay)
+    }
+    
+    override func draw(_ mapRect: MKMapRect, zoomScale: MKZoomScale, in context: CGContext) {
+        guard let imageRef = overlayImage.cgImage else {
+            return
+        }
+        let theMapRect = overlay.boundingMapRect
+        let theRect = rect(for: theMapRect)
+        
+        context.scaleBy(x: 1, y: -1)
+        context.translateBy(x: 0, y: -theRect.height)
+        context.draw(imageRef, in: theRect)
+    }
+}
+
+
+extension MapViewController: MKMapViewDelegate {
+    
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        // If annotation is not of type RestaurantAnnotation (MKUserLocation types for instance), return nil
+        if !(annotation is BurnAnnotation){
```

Comment:
here is a great use case for a guard statement. if we instead have

```swift
guard let burnAnnotation = annotation as? BurnAnnotation else {
    return nil
}
```

Then we can use it safely in the rest of the function without any force casts